src/commons_codec/model.py

```diff
@@ -81,3 +82,33 @@ def from_json(cls, payload: str) -> t.Union["ColumnTypeMapStore", None]:
         if not payload:
             return None
         return cls.from_dict(json.loads(payload))
+
+
+@define
+class SQLOperation:
+    """
+    Bundle data about an SQL operation, including statement and parameters.
+
+    Parameters can be a single dictionary or a list of dictionaries.
+    """
+
+    statement: str
+    parameters: t.Optional[t.Union[t.Mapping[str, t.Any], t.List[t.Mapping[str, t.Any]]]] = None
+
+
+@define
+class SQLParameterizedClause:
+    """
+    Manage details about a SQL parameterized clause, including column names, parameter names, and values.
+    """
+
+    columns: t.List[str] = Factory(list)
+    names: t.List[str] = Factory(list)
+    values: t.List[str] = Factory(list)
+
+    @property
+    def set_clause(self):
+        """
+        Render a SET clause of an SQL statement.
+        """
+        return ", ".join([f"{key}={value}" for key, value in zip(self.columns, self.names)])
```

Comment:
Thanks for looking at the details. `self.columns` is used right here to render the parameterized `SET` clause.